src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -109,22 +102,30 @@ async def deploy_and_assert_grafana_agent(
     await model.wait_for_idle(apps=[GRAFANA_AGENT_APP], status="blocked", timeout=5 * 60)
 
 
-async def _check_metrics_endpoint(app: Application, metrics_endpoint: str) -> None:
-    """Check metrics endpoint accessibility.
+def _check_url(url: str, port: int, path: str) -> bool:
+    """Check port and path in url."""
+    output = urlparse(url)
+    return output.port == port and output.path == path
 
-    Checking accessibility of metrics endpoint from grafana-agent-k8s. If metrics endpoint is
-    defined as `*:5000/metrics` it will be changed to `<app-name>.<namespace>.svc:5000/metrics`.
+
+async def _check_metrics_endpoint(app: Application, port: int, path: str) -> None:
+    """Check metrics targets accessibility.
+
+    Checking if metrics port and metrics path are dedine in any targes of Grafana agent.
+    E.g. curl localhost:12345/agent/api/v1/metrics/targets return with success status and it's
+    configured as expected.
```

Comment:
Can we be more explicit about what we are actually expecting the return of the curl to be? What are the bits of that output we are interested in and how that can help us with the tests.

Reply:
I will privide short example in docstring and description about which key we need. 👍